python/ray/util/sgd/torch/examples/segmentation/train_segmentation.py

```diff
@@ -102,6 +80,67 @@ def criterion(inputs, target):
 
 
 class SegOperator(TrainingOperator):
+    def setup(self, config):
+        args = config["args"]
+        # Create Data Loaders.
+        with RayFileLock():
+            # Within a machine, this code runs synchronously.
+            dataset, num_classes = get_dataset(
+                args.dataset, "train", get_transform(train=True))
+            config["num_classes"] = num_classes
+            dataset_test, _ = get_dataset(
+                args.dataset, "val", get_transform(train=False))
+            data_loader = torch.utils.data.DataLoader(
+                dataset,
+                batch_size=args.batch_size,
+                num_workers=args.data_workers,
+                collate_fn=utils.collate_fn,
+                drop_last=True)
+
+            data_loader_test = torch.utils.data.DataLoader(
+                dataset_test,
+                batch_size=1,
+                num_workers=args.data_workers,
+                collate_fn=utils.collate_fn)
+
+        # Create model.
+        model = torchvision.models.segmentation.__dict__[args.model](
+            num_classes=config["num_classes"],
+            aux_loss=args.aux_loss,
+            pretrained=args.pretrained)
+        if config["num_workers"] > 1:
+            model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+
+        # Create optimizer.
+        params_to_optimize = [
+            {
+                "params": [
+                    p for p in model.backbone.parameters() if p.requires_grad
+                ]
+            },
+            {
+                "params": [
+                    p for p in model.classifier.parameters() if p.requires_grad
+                ]
+            },
+        ]
+        if args.aux_loss:
+            params = [
+                p for p in model.aux_classifier.parameters() if p.requires_grad
+            ]
+            params_to_optimize.append({"params": params, "lr": args.lr * 10})
+        optimizer = torch.optim.SGD(
+            params_to_optimize,
+            lr=args.lr,
+            momentum=args.momentum,
+            weight_decay=args.weight_decay)
```

Comment:
how about still keeping this in a function?

Reply:
makes sense, changed